modules/core/src/main/java/org/apache/ignite/internal/client/thin/TcpIgniteClient.java

```diff
@@ -114,6 +114,7 @@ private TcpIgniteClient(ClientConfiguration cfg) throws ClientException {
         binary = new ClientBinary(marsh);
 
         ch = new ReliableChannel(chFactory, cfg, binary);
+        ch.initConnection();
```

Comment:
Is there a reason to bring this outside of the constructor?

Reply:
`initCollection` perform async operations, so it will lead to escaping incomplete ReliableChannel. I'm not confident is there too much troubles in our particular case, but best practice is to avoid such situations. IMHO, lazy initialization is preferable but there are too much test cases that rely that connection is initialized from beginning. So let's do not break the API, but preserve this object safe.